sky/shell/ui/engine.cc

```diff
@@ -96,27 +96,18 @@ void Engine::Init() {
   blink::initialize(g_platform_impl);
 }
 
-void Engine::BeginFrame(base::TimeTicks frame_time) {
+std::unique_ptr<LayerTree> Engine::BeginFrame(base::TimeTicks frame_time) {
   TRACE_EVENT0("sky", "Engine::BeginFrame");
 
+  std::unique_ptr<LayerTree> layer_tree;
   if (sky_view_)
-    sky_view_->BeginFrame(frame_time);
-}
-
-PassRefPtr<SkPicture> Engine::Paint() {
-  TRACE_EVENT0("sky", "Engine::Paint");
-
-  if (sky_view_) {
-    RefPtr<SkPicture> picture = sky_view_->Paint();
-    if (picture)
-      return picture.release();
-  }
+    layer_tree = sky_view_->BeginFrame(frame_time);
+  if (!layer_tree)
+    layer_tree.reset(new LayerTree());
```

Comment:
there is std::make_unique() but that's c++14, not sure if you want to go that far into the future